crates/brainfuck_prover/src/components/memory/table.rs

```diff
@@ -0,0 +1,225 @@
+use num_traits::{One, Zero};
+use stwo_prover::core::fields::m31::BaseField;
+
+/// Represents a single row in the Memory Table.
+///
+/// The Memory Table stores:
+/// - The clock cycle counter (`clk`),
+/// - The memory pointer (`mp`),
+/// - The memory value (`mv`),
+/// - The dummy column flag (`d`).
+#[derive(Debug, Default, PartialEq, Eq, Clone)]
+pub struct MemoryTableRow {
+    /// Clock cycle counter: the current step.
+    pub clk: BaseField,
+    /// Memory pointer: points to a memory cell.
+    pub mp: BaseField,
+    /// Memory value: value of the cell pointer by `mp` - values in [0..2^31 - 1)
+    pub mv: BaseField,
+    /// Dummy: Flag whether the row is a dummy one or not
+    d: BaseField,
+}
+
+impl MemoryTableRow {
+    pub fn new(clk: BaseField, mp: BaseField, mv: BaseField) -> Self {
+        Self { clk, mp, mv, d: BaseField::zero() }
+    }
+
+    pub fn new_dummy(clk: BaseField, mp: BaseField, mv: BaseField) -> Self {
+        Self { clk, mp, mv, d: BaseField::one() }
+    }
+}
+
+/// Represents the Memory Table, which holds the required registers
+/// for the Memory component.
+///
+/// The Memory Table is constructed by extracting the required fields
+/// from the execution trace provided by the Brainfuck Virtual Machine,
+/// then sorting by `mp` as a primary key and by `clk` as a secondary key.
+///
+/// To enforce the sorting on clk, all clk jumped are erased by adding dummy rows.
+/// A dummy column flags them.
+#[derive(Debug, Default, PartialEq, Eq, Clone)]
+pub struct MemoryTable {
+    /// A vector of [`MemoryTableRow`] representing the table rows.
+    pub table: Vec<MemoryTableRow>,
+}
+
+impl MemoryTable {
+    /// Creates a new, empty [`MemoryTable`].
+    ///
+    /// # Returns
+    /// A new instance of [`MemoryTable`] with an empty table.
+    pub const fn new() -> Self {
+        Self { table: Vec::new() }
```

Comment:
```suggestion
        Self::default()
```

Reply:
Oh, didn't catch it indeed
We've got the same in `instruction/table.rs`

Reply:
However, it requires not having the new method as const (or implement Default as const ?)